vhpc_toolkit/config_objs.py

```diff
@@ -974,6 +974,40 @@ def create_svs(self, svs_name, vmnic, num_ports=8):
         host_network_obj = self.host_obj.configManager.networkSystem
         host_network_obj.AddVirtualSwitch(vswitchName=svs_name, spec=svs)
 
+    def modify_sriov(
+        self,
+        device_id: str,
+        num_virtual_functions: int = None,
+        enable_sriov: bool = True,
+    ):
+        """
+        Function to enable/disable SRIOV and/or change the number of virtual functions on devices of a host.
+        Number of virtual functions argument is skipped if user is trying to disable SRIOV
+        Args:
+            device_id: Address of the device which supports SRIOV in format xxxx:xx:xx.x
+            num_virtual_functions: Number of virtual functions to set for the NIC
+            enable_sriov: Whether to enable or disable SRIOV for the NIC
+
+        Returns:
+            None
+        """
+        config = vim.host.SriovConfig()
+        config.sriovEnabled = enable_sriov
+        if enable_sriov and num_virtual_functions:
+            config.numVirtualFunction = num_virtual_functions
+        config.id = device_id
+        try:
+            self.host_obj.configManager.pciPassthruSystem.UpdatePassthruConfig(
+                config=[config]
+            )
+            self.logger.info(
+                f"{'enabled' if enable_sriov else 'disabled'} SRIOV for network with location: {device_id} on host {self.host_obj.name}"
```

Comment:
SRIOV for network with location --> SRIOV on network adapter with PCIe Device ID